assigner.py

```diff
@@ -35,24 +37,119 @@ def main():
         sys.exit(127)
+
+    # Bugzilla setup
     bcfg = config.get('bugzilla')
-    bapi = bugzilla.Bugzilla(url=bcfg.get('url'), api_key=os.environ.get('BUGZILLA_API_KEY'))
+    bapi_key = os.environ.get('BUGZILLA_API_KEY')
+    if (bapi_key is None):
+        logger.critical("No Bugzilla API Key passed in environment variable BUGZILLA_API_KEY")
+        sys.exit(127)
+    bapi = bugzilla.Bugzilla(url=bcfg.get('url'), api_key=bapi_key)
+
+    # Casa setup
+    ccfg = config.get('casa')
+    capi_key = os.environ.get('CASA_API_KEY')
+    if (capi_key is None):
+        logger.critical("No CASA API Key passed in environment variable CASA_API_KEY")
+        sys.exit(127)
+    capi = casa.Casa(url=ccfg.get('url'), api_key=capi_key)
 
+    # Do things!
     autoassign(bapi, bcfg.get('rra'), args.dry_run)
     autoassign(bapi, bcfg.get('va'), args.dry_run)
+    autocasa(bapi, capi, bcfg, ccfg, args.dry_run)
 
+def autocasa(bapi, capi, bcfg, ccfg, dry_run):
+    """
+    This will search through closed bugs and update CASA accordingly.
+    @bcfg: bugzilla configuration dict
+    @ccfg: casa configuration dict
+    """
+
+    bcfg_va = bcfg.get('va')
+    bcfg_rra = bcfg.get('rra')
+
+    # Look for bugs that are up to ccfg.lookup_period_in_days days old
+    lookup_period = (datetime.now() - timedelta(days=ccfg.get('lookup_period_in_days'))).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    # Look for all registered products, for that lookup_period
+    terms = [{'product': bcfg_va.get('product')}, {'product': bcfg_rra.get('product')},
+             {'component': bcfg_va.get('component')}, {'component': bcfg_rra.get('component')},
+             {'status': 'RESOLVED'}, {'status': 'VERIFIED'},
+             {'last_change_time': lookup_period},
+             {'creator': ccfg.get('bot_email')}
+            ]
+    bugs = bapi.search_bugs(terms)['bugs']
+    logger.debug('Analyzing {} closed bugs...'.format(len(bugs)))
+
+    for bug in bugs:
+        # Get casa project id and other metadata
+        comments = bapi.get_comments(bug.get('id'))['bugs'][str(bug.get('id'))]['comments']
+        casa_data = capi.parse_casa_comment(comments[0]['text'])
+        project_id = casa_data.get('project_id')
+        if (len(casa_data)) == 0:
+            logger.warning("Could not find any CASA data in comment 0 even thus this comment was created by CASA!")
+            continue
+
+        # Some basic checks that we can update that project
+        ## Have bugzilla support?
+        casa_project = capi.casa_get_project(casa_data.get('project_id'))
+        if casa_project['syncedToIntegrations']['bugzilla'] is not True:
+            logger.warning('Project {} has no bugzilla integration, skipping!'.format(project_id))
+            continue
+
+        # Is already approved/disapproved in some way?
+        ## XXX This means Bugzilla cannot override a status already set, thus, if you set "WONTFIX" in bugzilla,
```

Comment:
Was there intent to fill something in for "XXX"?

Reply:
`XXX` is like `TODO` its a marker that shows up in many editor